firefox-ios/Client/Frontend/TrackingProtection/TrackingProtectionViewController.swift

```diff
@@ -190,20 +189,36 @@ class TrackingProtectionViewController: UIViewController,
     // MARK: Redux
     func newState(state: TrackingProtectionState) {
         trackingProtectionState = state
-        if trackingProtectionState.showTrackingProtectionSettings {
-            enhancedTrackingProtectionMenuDelegate?.settingsOpenPage(settings: .contentBlocker)
-        } else if trackingProtectionState.showDetails {
-            showTrackersDetailsController()
-        } else if trackingProtectionState.showBlockedTrackers {
-            showBlockedTrackersController()
-        } else if trackingProtectionState.showsClearCookiesAlert {
-            onTapClearCookiesAndSiteData()
-        } else if trackingProtectionState.shouldClearCookies {
+        if let navigateTo = state.navigateTo {
+            switch navigateTo {
+            case .home:
+                navigationController?.popToRootViewController(animated: true)
+            case .back:
+                navigationController?.popViewController(animated: true)
+            case .close:
+                enhancedTrackingProtectionMenuDelegate?.didFinish()
+            case .settings:
+                enhancedTrackingProtectionMenuDelegate?.settingsOpenPage(settings: .contentBlocker)
+            }
```

Comment:
since this will be addressed in a separate PR, should we put a TODO instead here instead of implementing the navigation logic?

Reply:
I would add a TODO but removing the navigation code would render the navigation bar unresponsive until the other PR is merged, is that preferred?

Reply:
I think that's okay and we can add a `TODO` since this feature is still a work in progress + so we merge code that we're confident that we want to keep in.